Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -74,7 +74,7 @@ public void NotThrow(string because = "", params object[] becauseArgs)
         {
             try
             {
-                Task.Run(Subject).Wait();
+                Subject().Wait();
```

Comment:
Shouldn't you use the `ExecuteInDefaultSynchronizationContext` method in all the below changes?

Reply:
I don't believe we need that, based on the results I'm seeing. 
According to AppVeyor, the only hanging tests on each of the my failing builds were:

- FluentAssertions.Specs.AsyncFunctionExceptionAssertionSpecs.When_async_method_throws_exception_expected_not_to_be_thrown_it_should_fail

- FluentAssertions.Specs.FluentActionsSpecs.Awaiting_works_with_func

What both tests have in common is that they both use Awaiting extension methods. 

Awaiting_works_with_func uses:

FluentActions.cs 
`public static Func<Task<T>> Awaiting<T>(Func<Task<T>> func)`

When_async_method_throws_exception_expected_not_to_be_thrown_it_should_fail uses:

AssertionExtensions.cs
`public static Func<Task> Awaiting<T>(this T subject, Func<T, Task> action)`

I believe these methods, combined with no longer using Task.Run in the NotThrow method, were causing the tests to hang. 
I've made a change to Awaiting in FluentActions.cs and the tests are passing fine. Interestingly, I haven't modified the Awaiting method in AssertionExtensions.cs, but the the related test is fine too. At this stage I'm not entirely sure why. It might be that this is a transient issue which only shows up on certain test runs. 

Maybe we can somehow re-run the tests a few times on AppVeyor and see if it manifests itself again?

Reply:
I'm inclined to remove the synchronization context for all invocations to an `Action` and `Func<Task>`. See also this issue reported on Chill. https://github.com/ChillBDD/Chill/issues/77

Reply:
I'm inclined to agree with you on this. I want to dig around a bit first though and try to determine categorically why specific tests are hanging and which ones aren't. 
I've created a separate branch and PR just for diagnosing these hanging tests in AppVeyor. I'll reply here and update this PR when I find out more.

Reply:
Kudos for helping us solve this long-lasting problem.

Reply:
@dennisdoomen No worries, I'm just happy to try contributing to this fantastic project :) 

I made the changes you suggested, by adding ExecuteInDefaultSynchronizationContext in various places. I've built these changes several times on AppVeyor and the tests seem stable. 

I'll leave you to resolve this if you're satisfied, otherwise let me know if you have other suggestions.